src/pages/api/newsletter/subscribe.js

```diff
@@ -7,15 +7,15 @@ export default async function handler(req, res) {
     if (req.method === 'POST') {
       // Validate request body
       if (!req.body.email) {
-        res.status(400).send('Missing email');
+        res.status(400).send({ message: 'Missing email', status: 400 });
         return;
       }
 
       // Validate email
       const emailRegex =
         /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
       if (!emailRegex.test(req.body.email)) {
-        res.status(400).send('Invalid email');
+        res.status(400).send({ message: 'Invalid Email Address', status: 400 });
```

Comment:
When I tested and passed `asdf` or anything that is not an email address,  as an email address, I got `An error occurred` instead of the above message.

Reply:
You tested locally? So, you probably don't have access to the env variable

Reply:
So, it didn't even get in. it stopped at where we checked if some env variables are available.

Reply:
I didn't test locally. I tested with the preview link Vercel bot provides for every opened PR above

Reply:
Okay, same though.LOL no env variable that's why @vyktoremario .

Reply:
Oh okay.
Makes sense